### PR TITLE
Itemdb Racing hat and Racing enchant

### PR DESCRIPTION
**Pull request by attackjom**
* **Addressed Issue(s)**: - 

* **Server Mode**:  Renewal

* **Description of Pull Request**: 

- Add Racing cap and their enchant itemDB
- Too many item has a unknow rates or specialeffect on autobonus script that need to correct in the future.
- All of itemscript make from translation in divine-pride.net (Thanks to "Sigma")
https://www.divine-pride.net/forum/index.php?/topic/3266-hugel-monster-racing/

### CODE REVIEW

File: db/re/item_db.txt [hunk 1]
```diff
@@ -10158,6 +10158,23 @@
...
+19191,Racing_Cap_GC,Racing Cap (Guillotine Cross),4,20,,0,,10,,1,0x00001000,56,2,256,,100,1,1134,{ .@r = getrefine(); bonus bBaseAtk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) bonus bCritAtkRate,5*getskilllv("AS_LEFT"); if (.@r>=11) autobonus "{ bonus bCritAtkRate,20; }",50,5000,BF_WEAPON; /* Unknow Rate and Effect */ },{},{}
+19192,Racing_Cap_SC,Racing Cap (Shadow Chaser),4,20,,0,,10,,1,0x00020000,56,2,256,,100,1,1134,{ .@r = getrefine(); bonus bMatk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) bonus bMatk,5*getskilllv("RG_PLAGIARISM"); if (.@r>=11) bonus3 bAutoSpell,"HW_MAGICPOWER",5,30; /* Unknow Rates */ },{},{}
+19193,Racing_Cap_WL,Racing Cap (Warlock),4,20,,0,,10,,1,0x00000200,56,2,256,,100,1,1134,{ .@r = getrefine(); bonus bMatk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) bonus2 bSkillCooldown,"WL_TETRAVORTEX",-500*(getskilllv("WZ_METEOR")+getskilllv("WZ_VERMILION")); if (.@r>=11) bonus2 bVariableCastrate,"HW_MAGICPOWER",-50; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: HW_MAGICPOWER only has fixed cast time. `bFixedCastrate` should be used instead.

File: db/re/item_db.txt [hunk 2]
```diff
@@ -10158,6 +10158,23 @@
...
+19194,Racing_Cap_SO,Racing Cap (Sorcerer),4,20,,0,,10,,1,0x00010000,56,2,256,,100,1,1134,{ .@r = getrefine(); bonus bMatk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) { bonus2 bSkillAtk,"MG_THUNDERSTORM",5*getskilllv("SA_AUTOSPELL"); bonus2 bSkillAtk,"MG_FIREBALL",5*getskilllv("SA_AUTOSPELL"); } if (.@r>=11) { bonus2 bSkillAtk,"MG_FIREBOLT",50; bonus2 bSkillAtk,"MG_LIGHTNINGBOLT",50; } },{},{}
+19195,Racing_Cap_RK,Racing Cap (Rune Knight),4,20,,0,,10,,1,0x00000080,56,2,256,,100,1,1134,{ .@r = getrefine(); bonus bBaseAtk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) bonus bCritAtkRate,5*getskilllv("KN_TWOHANDQUICKEN"); if (.@r>=11) { bonus bUnbreakableWeapon; skill "TF_DOUBLE",5; bonus bDoubleRate,25; } },{},{}
+19196,Racing_Cap_RG,Racing Cap (Royal Guard),4,20,,0,,10,,1,0x00004000,56,2,256,,100,1,1134,{ .@r = getrefine(); bonus bBaseAtk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) bonus2 bSkillAtk,"LG_RAGEBURST",5*getskilllv("KN_SPEARMASTERY"); if (.@r>=11) bonus2 bSkillCooldown,"LG_EARTHDRIVE",-1000; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: LG_RAGEBURST damage bonus should be 10% per KN_SPEARMASTERY level.

File: db/re/item_db.txt [hunk 3]
```diff
@@ -10184,6 +10201,8 @@
...
 19391,Eyes_Of_Illusion,Eyes Of Illusion,4,20,,500,,2,,0,0xFFFFFFFF,63,2,512,,100,0,1779,{ bonus bMaxHPrate,10; bonus bMaxSPrate,10; bonus2 bSubRace,RC_Player,5; bonus2 bResEff,Eff_Blind,10000; if (getskilllv("GN_ILLUSIONDOPING")==5) skill "SA_DISPELL",5; if (getskilllv("GN_MANDRAGORA")==5) bonus2 bFixedCastrate,"GN_MANDRAGORA",-70; },{},{}
+19396,Racing_Cap_SG,Racing Cap (Star Gladiator),4,20,,0,,10,,1,0x00400000,63,2,256,,100,1,1134,{ .@r = getrefine(); bonus bBaseAtk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) bonus bBaseatk,5*getskilllv("TK_HPTIME"); if (.@r>=11) autobonus3 "{ bonus2 bSkillAtk,\"SJ_SOLARBURST\",30; }",1000,10000,"SJ_PROMINENCEKICK"; },{},{}
+19397,Racing_Cap_SL,Racing Cap (Soul Linker),4,20,,0,,10,,1,0x00800000,63,2,256,,100,1,1134,{ .@r = getrefine(); bonus bMatk,10*(.@r/2); bonus bAspdRate,2*(.@r/3); if (.@r>=8) bonus bMaxHP,100*getskilllv("TK_SPTIME"); if (.@r>=11) autobonus3 "{ bonus2 bSkillAtk,\"SP_SPA\",30; bonus2 bSkillAtk,\"SP_SWHOO\",30; }",1000,10000,"SL_SMA"; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: `bonus bMaxHP,100*getskilllv("TK_SPTIME");` should be `bMaxSP`.

File: db/re/item_combo_db.txt [hunk 4]
```diff
@@ -793,6 +793,58 @@
...
+19190:29320,{ autobonus3 "{ bonus2 bAddSize,Size_All,7; }",1000,60000,"GN_CARTBOOST"; bonus2 bSkillCooldown,"GN_MANDRAGORA",-1000; }
+19190:29321,{ autobonus3 "{ bonus bBaseAtk,40; bonus2 bAddSize,Size_All,15; }",1000,60000,"GN_CARTBOOST"; bonus2 bSkillCooldown,"GN_MANDRAGORA",-5000; }
+19191:29322,{ autobonus3 "{ bonus bBaseAtk,20; }",1000,60000,"ASC_EDP"; }
+19191:29323,{ autobonus3 "{ bonus bBaseAtk,30; }",1000,60000,"ASC_EDP"; bonus2 bAddSize,Size_All,5; }
+19191:29324,{ autobonus3 "{ bonus bBaseAtk,50; bonus bCritical,10; }",1000,60000,"ASC_EDP"; bonus2 bAddSize,Size_All,10; }
+19192:29325,{ autobonus3 "{ bonus bCritical,5; }",1000,60000,"SC_AUTOSHADOWSPELL"; }
+19192:29326,{ autobonus3 "{ bonus bCritical,10; }",1000,60000,"SC_AUTOSHADOWSPELL"; bonus2 bMagicAtkEle,Ele_Water,5; bonus2 bMagicAtkEle,Ele_Fire,5; bonus2 bMagicAtkEle,Ele_Wind,5; bonus2 bMagicAtkEle,Ele_Earth,5; }
+19192:29327,{ autobonus3 "{ bonus bCritical,20; bonus bMatk,30; }",1000,60000,"SC_AUTOSHADOWSPELL"; bonus2 bMagicAtkEle,Ele_Water,15; bonus2 bMagicAtkEle,Ele_Fire,15; bonus2 bMagicAtkEle,Ele_Wind,15; bonus2 bMagicAtkEle,Ele_Earth,15; }
+19193:2000,{ bonus2 bVariableCastrate,"HW_MAGICPOWER",50; }
```
Comment (secretdataz) [inline db/re/item_combo_db.txt]: unintended?

Comment (attackjom) [reply db/re/item_combo_db.txt]: It has a description on bottom of description so I will change from varcast -> fixcast.

File: db/re/item_db.txt (hunk 3 above)
Comment (attackjom) [reply db/re/item_db.txt]: done

File: db/re/item_db.txt (hunk 2 above)
Comment (attackjom) [reply db/re/item_db.txt]: done

File: db/re/item_db.txt (hunk 1 above)
Comment (attackjom) [reply db/re/item_db.txt]: done

**Pull request merged**